WordPress/gutenberg
# Dropdown: Fix `testing-library/no-node-access` in tests

## What?
With the recent work to improve the quality of tests, we fixed a bunch of ESLint rule violations. This PR fixes a few [`no-node-access`](https://github.com/testing-library/eslint-plugin-testing-library/blob/main/docs/rules/no-node-access.md) rule violations in the `Dropdown` component. 

## Why?
The end goal is to enable that ESLint rule once all violations have been fixed.

## How?
Replace all `querySelector` and `querySelectorAll` with `*ByRole` and `*ByText`.

## Testing Instructions
Verify all tests still passes by running `npm run test:unit -- packages/components/src/dropdown/`
